tests/test_outputs/test_form_output_prep.py

```diff
@@ -0,0 +1,361 @@
+            [39900000408, 0, 202212, 'AA', np.nan, np.nan, 100.0, 'No', 1.0, 6,
+                40, 2022],
+            [39900000408, 1, 202212, 'AA', 0.0, 0.0, 100.0, 'No', 1.0, 6,
+                40, 2022],
+            [39900000408, 2, 202212, 'AA', 0.0, 0.0, 100.0, 'No', 1.0, 6,
+                  40, 2022],
+            [39900000576, 1, 202212, 'AA', 0.0, 0.0, 100.0, np.nan, 1.0, 1,
+                40, 2022],
+            [39900000960, 0, 202212, 'AA', np.nan, np.nan, np.nan, 'No', 1.0, 1, 40,
+                2022],
+            [39900000960, 1, 202212, 'AA', np.nan, 0.0, np.nan, 'No', 1.0, 1, 40,
+                2022],
+            [39900001029, 1, 202212, 'I', 119376.5, 244543.1667, 100.0,
+                np.nan, 1.0, 1, 14, 2022]
+        ]
+        full_outputs = pd.DataFrame(data=data, columns=columns)
+        return full_outputs
+    
+
+    @pytest.fixture(scope="function")
+    def tau_expected(self) -> pd.DataFrame:
+        columns = [
+            'reference',
+            'instance',
+            'period',
+            '201',
+            '202',
+            '211',
+            '602',
+            '604',
+            'a_weight',
+            'formtype',
+            'pg_numeric',
+            'period_year',
+            'ultfoc',
+            '407',
+            'rusic',
+            'form_status']
+        data = [
+            [39900001031, 1, 202212.0, 'U', 0.0, 0.0, 100.0, 'No', 49.0, 6,
+                30.0, 2022.0, np.nan, np.nan, np.nan, np.nan],
+            [39900001031, 2, 202212.0, 'U', 0.0, 0.0, 100.0, 'No', 49.0, 6,
+                30.0, 2022.0, np.nan, np.nan, np.nan, np.nan],
+            [39900001032, 1, 202212.0, 'I', 119376.5, 244543.16666666663,
+                100.0, np.nan, 1.0, 1, 14.0, 2022.0, np.nan, np.nan, np.nan, np.nan],
+            [39900001577, 1, np.nan, 'C', np.nan, np.nan, 100.0, 'Yes', 1.0, "0003", 3.0,
+                np.nan, 'GB ', 0.72, 10130.0, 600.0],
+            [39900006060, 1, np.nan, 'G', np.nan, np.nan, 100.0, 'Yes', 1.0, "0003", 12.0,
+                np.nan, 'GB ', 0.6, 20120.0, 600.0],
+            [39900008752, 1, np.nan, 'AA', np.nan, np.nan, 100.0, 'Yes', 1.0, "0003", 40.0,
+                np.nan, 'GB ', 0.72, 46420.0, 600.0],
+            [39900008762, 1, np.nan, np.nan, np.nan, np.nan, 100.0, 'Yes', 1.0, "0003", np.nan,
+                np.nan, 'GB ', 6.9, 22230.0, 600.0]
+        ]
+        full_outputs = pd.DataFrame(data=data, columns=columns)
+        return full_outputs
+
+
+    def test_form_output_prep_with_ni(self,
+                                    estimated_df,
+                                    weighted_df,
+                                    ni_full_responses,
+                                    sic_pg_num,
+                                    pg_num_alpha,
+                                    ni_expected,
+                                    full_output_expected,
+                                    tau_expected):
```

Comment:
Our pipeline style would have the formatting as:

    def test_form_output_prep_with_ni(
        self,
        estimated_df,
        weighted_df,
        ni_full_responses,
        sic_pg_num,
        pg_num_alpha,
        ni_expected,
        full_output_expected,
        tau_expected
    ):

Reply:
please could you update throughout? Thanks.

Reply:
Completely agree with this. A miss from me, will update this.

Reply:
done